### PR TITLE
Implementing changes from Matrix project.

### PR DESCRIPTION
**Pull request by DmytroLohvynenko**
Ref https://github.com/CluedIn-io/CluedIn.Connector.Matrix/issues/98

Some test setup code was disabled.

### CODE REVIEW

File: src/Connector.PowerBI/Connector/PowerBIConnector.cs [hunk 1]
```diff
@@ -36,21 +27,28 @@ public override async Task CreateContainer(ExecutionContext executionContext, Gu
...
+
+                var table = new PushDataset()
                 {
                     DefaultMode = "Push",
                     Name = model.Name,
                     Tables = new List<Table>() { new Table() { Name = model.Name, Columns = columns.Select(i => new Column() { Name = i.Name, DataType = i.Type.ToString() }).ToList() } }
-                });
+                };
+
+                // TODO: Check the table is sended correctly
```
Comment (andrew-hardy) [inline src/Connector.PowerBI/Connector/PowerBIConnector.cs:40]: Has this TODO been done? If so please remove it

File: src/Connector.PowerBI/Connector/PowerBIConnector.cs [hunk 2]
```diff
@@ -290,30 +243,19 @@ private string GetDbType(VocabularyKeyDataType type)
...
         }
 
         public override Task EmptyContainer(ExecutionContext executionContext, Guid providerDefinitionId, string id)
         {
             return Task.CompletedTask;
         }
+
+        public override Task<bool> VerifyConnection(ExecutionContext executionContext, IDictionary<string, object> authenticationData)
+        {
+            throw new NotImplementedException();
```
Comment (andrew-hardy) [inline src/Connector.PowerBI/Connector/PowerBIConnector.cs:258]: Should this be implemented?

File: test/unit/Connector.PowerBI.Test/TestContext.cs [hunk 3]
```diff
@@ -77,106 +77,106 @@ public TestContext() : this(Mock.Of<ILogger>())
...
+            //Bus.Setup(s => s.IsConnected).Returns(false);
+            //Bus.Setup(s => s.Advanced.IsConnected).Returns(false);
+
+            //OrganizationFactory = (ctx, id) => new TestOrganization(ctx.ApplicationContext, id);
+
+            //var o1 = Container.Resolve<ApplicationContext>();
+
+            //OrganizationRepository.Setup(r => r.GetOrganization(It.IsAny<ExecutionContext>(), new TestOrganization(o1, Constants.SystemOrganizationId).Id)).Returns<ExecutionContext, Guid>((c, i) => OrganizationFactory(c, i));
+
+            //OrganizationRepository.Setup(r => r.GetOrganization(It.IsAny<ExecutionContext>(), It.IsAny<Guid>())).Returns<ExecutionContext, Guid>((c, i) => OrganizationFactory(c, i));
```
Comment (andrew-hardy) [inline test/unit/Connector.PowerBI.Test/TestContext.cs:179]: Please uncomment this and fix test if required

**Review by andrew-hardy — CHANGES_REQUESTED** [review]

File: src/Connector.PowerBI/Connector/PowerBIConnector.cs (hunk 2 above)
Comment (DmytroLohvynenko) [reply src/Connector.PowerBI/Connector/PowerBIConnector.cs:258]: before it was mocked to always return `true`. 
Unfortunately we haven't had enough time and PowerBI account to test this connector.

File: test/unit/Connector.PowerBI.Test/TestContext.cs (hunk 3 above)
Comment (DmytroLohvynenko) [reply test/unit/Connector.PowerBI.Test/TestContext.cs]: there are no active tests in the solution <>
This was some kind of test setup, but actually not used anywhere (or I can't find a place where it is used).
So just removed it.

File: src/Connector.PowerBI/Connector/PowerBIConnector.cs (hunk 1 above)
Comment (DmytroLohvynenko) [reply src/Connector.PowerBI/Connector/PowerBIConnector.cs:40]: Probably the same as in other case.
It was added, but connector wasn't in priority to check.

File: src/Connector.PowerBI/Connector/PowerBIConnector.cs (hunk 2 above)
Comment (DmytroLohvynenko) [reply src/Connector.PowerBI/Connector/PowerBIConnector.cs]: returned mocked version, now always return `true`.

File: src/Connector.PowerBI/Connector/PowerBIConnector.cs (hunk 1 above)
Comment (DmytroLohvynenko) [reply src/Connector.PowerBI/Connector/PowerBIConnector.cs]: removed TODO comment.

**Review by romaklimenko — APPROVED** [review]

**Review by andrew-hardy — APPROVED** [review]

**Pull request merged**